src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Engine/AbstractFormatEngine.cs

```diff
@@ -162,13 +162,17 @@ protected virtual NodeOperations CreateNodeOperations(CancellationToken cancella
             return nodeOperations;
         }
 
-        private static void AddOperations<T>(List<T> operations, List<T> scratch, SyntaxNode node, Action<List<T>, SyntaxNode> addOperations)
+        private static void AddOperations<T>(SegmentedList<T> operations, List<T> scratch, SyntaxNode node, Action<List<T>, SyntaxNode> addOperations)
```

Comment:
should Scratch also be a SegmentedList?

Reply:
Maybe? But I don't think it's obvious. I don't actually see that large of allocations for these in the profiles (they are there, but there are only a couple megs of allocations for them). I'll keep them as is for now unless you feel strongly.